0003-configuration-management-extension/README.md

```diff
@@ -0,0 +1,132 @@
+---
+name: Configuration Management Extension
+shortname: 3/CMX
+status: raw
+editor: Dmitry Sergeev <dsergeev@cybervisiontech.com>
+---
+
+## Introduction
+
+The Configuration Management Extension is a [Kaa Protocol](/0002-kaa-protocol/README.md) extension.
+
+It is intended to manage endpoint configuration distribution.
+
+## Requirements and constraints
+### Problems and possible solutions
+
+1. _Multiple configurations available for client._ It is possible that there will be more than one available configurations for endpoint (e.g.: endpoint was offline while new configurations were added).
+   
+   Solution:
+   - Only latest configuration should be sent to an endpoint to reduce amount of traffic.
+_Note:_ Approach for delivering firmware updates over the air should be different.
+
+2. _The server should know if a configuration has been delivered._ 
+
+   Solutions:
+   - Send PUBLISH message from client to "/status" topic. See below sections for details.
+
+## Use cases
+
+### UC1
+Configuration delivery by request. The endpoint should be able to receive latest configuration from CMX by request.
+
+### UC2
+Configuration delivery that is initiated by the server. The endpoint should receive latest configuration when it connects to a server if this configuration hadn't applied yet.
+
+## Design
+
+### Request/response
+The extension uses request/response pattern. A response from client is sent to confirm delivery.
+
+For MQTT, responses MUST be published at `<request_path>/status`. Each response MUST be published with the same QoS as the corresponding request.
+
+### Formats
+#### Schemeless JSON
+##### Use case 1
+The server should be able to listen for configuration request at the following resource path:
+```
+<endpoint_token>/config/request/json
+```
+
+The payload should be a JSON-encoded object with the following fields:
+- `id` (required) - id of the message. Should be either string or number. Used in delivery confirmation process.
```

Comment:
Not sure it should be required. Neither client, nor server require id to work properly.